auth/useAuthState.ts

```diff
@@ -7,14 +7,20 @@ export type AuthStateHook = LoadingHook<
   firebase.auth.Error
 >;
 
-export default (auth: firebase.auth.Auth): AuthStateHook => {
+export default (auth: firebase.auth.Auth, options: {onUserChanged?: (user?: firebase.User) => Promise<void>}): AuthStateHook => {
   const { error, loading, setError, setValue, value } = useLoadingValue<
     firebase.User | null,
     firebase.auth.Error
   >(() => auth.currentUser);
 
   useEffect(() => {
-    const listener = auth.onAuthStateChanged(setValue, setError);
+    const listener = auth.onAuthStateChanged(async (user) => {
+      if(typeof options?.onUserChanged === 'function') {
+         // onUserLoaded function to process custom claims on any other trigger function
+         await options.onUserChanged(user)
+      }
+      setValue(user);
```

Comment:
This will never be executed if the passed handler fails. It should be considered to set the value first and then perform the handler.

Reply:
@mauriceackel or should it be wrapped in try .. catch ?

Reply:
Alternatively it could be mentioned that the `onUserChanged` should handle errors within that function.

Reply:
This definitely needs some error handling built in, otherwise it will be very difficult to debug any issues. I'd propose a try / catch, then re-using `setError`